src/foam/nanos/dig/DUGRuleAction.js

```diff
@@ -49,64 +56,87 @@ foam.CLASS({
       name: 'applyAction',
       javaCode: `
       final var dugRule = (DUGRule) rule;
-      ((Logger) x.get("logger")).debug(this.getClass().getSimpleName(), "Sending DUG webhook", obj);
 
-      agency.submit(x, new ContextAgent() {
-        @Override
-        public void execute(X x) {
-          PM pm = new PM(getClass(), rule.getDaoKey(), rule.getName());
-          DAO dugDigestConfigDAO = (DAO) x.get("dugDigestConfigDAO");
-          DUGDigestConfig dugDigestConfig = (DUGDigestConfig) dugDigestConfigDAO.find(rule.getSpid());
-          try {
-            AbstractSink sink = null;
-            if ( dugDigestConfig != null && dugDigestConfig.getEnabled() ) {
-              sink = new HTTPDigestSink(
-                dugRule.getUrl(),
-                dugRule.evaluateBearerToken(),
-                dugDigestConfig,
-                dugRule.getFormat(),
-                new foam.lib.AndPropertyPredicate(
-                  x,
-                  new foam.lib.PropertyPredicate[] {
-                    new foam.lib.ExternalPropertyPredicate(),
-                    new foam.lib.NetworkPropertyPredicate(),
-                    new foam.lib.PermissionedPropertyPredicate()
-                  }
-                ),
-                true,
-                true
-              );
-            } else {
-              sink = new HTTPSink(
-                dugRule.getUrl(),
-                dugRule.evaluateBearerToken(),
-                dugRule.getFormat(),
-                new foam.lib.AndPropertyPredicate(
-                  x,
-                  new foam.lib.PropertyPredicate[] {
-                    new foam.lib.ExternalPropertyPredicate(),
-                    new foam.lib.NetworkPropertyPredicate(),
-                    new foam.lib.PermissionedPropertyPredicate()
-                  }
-                ),
-                true
-              );
+      // If the actingUser is set, we want to refind the obj with that user as the subject
+      // The goal with this is to filter the permissioned properties based on acting user
+      // Instead of the user that triggered the rule
+
+      final var actingUserId = getActingUser();
+
+      if ( actingUserId != 0 ) {
+        final var userDAO = (DAO) x.get("bareUserDAO");
+        final var user = (User) userDAO.find(actingUserId);
+
+        if ( user != null ) {
+          final var groupDAO = (DAO) x.get("localGroupDAO");
+          final var group = groupDAO.find(user.getGroup());
+
+          if ( group != null ) {
+            final var actingX = x.put("group", group);
+            final var objDAO = (DAO) actingX.get(rule.getDaoKey());
```

Comment:
DaoKey is often set to the local version of the DAO that is beyond the authentication layer or the PermissionedPropertyDAO. Can you add another property to the DUGRule that identifies the permissioned version of the DAO that should be used here instead of the lower level one?